### PR TITLE
print warning when there are ambiguous gatedefs or circuitdefs by def…

### PR DESCRIPTION
**Pull request by Spin1Half**
change default behavior to print a warning when there are ambiguous gate definitions or circuit definitions

### CODE REVIEW

**Comment by stylewarning** [comment]
I would change to emitting a `style-warning` instead of a full warning. You could make a new condition class for this, or just use `alexandria:simple-style-warning`.

That should fix the tests.

File: src/cl-quil.lisp [hunk 1]
```diff
@@ -44,6 +47,15 @@
 ;;; generally determined by a pattern matching system, and so we have to
 ;;; consider the arity and values of parameters and arguments).
 
+(defun warn-on-ambiguous-gate-or-circuit-definition (condition)
+  "Handler which prints a warning in the presence of a CONDITION indicating an ambiguous gate or circuit declaration."
+  (check-type condition ambiguous-definition-condition)
+  (let ((instr (ambiguous-definition-instruction condition)))
+    (typecase instr
+      (gate-definition (alexandria:simple-style-warning (format nil "Gate ~A has multiple definitions, this leads to ambiguous behavior.~%" (gate-definition-name instr))))
```
Comment (karlosz) [inline src/cl-quil.lisp:55]: You don't need to write `(format nil ...)` here. `warn` already can deal with a format string and format arguments by itself.

File: src/cl-quil.lisp [hunk 2]
```diff
@@ -44,6 +47,15 @@
...
+      (circuit-definition (alexandria:simple-style-warning (format nil "Circuit ~A has multiple definitions, this leads to ambiguous behavior.~%" (circuit-definition-name instr)))))
```
Comment (karlosz) [inline src/cl-quil.lisp:56]: Ditto.

**Review by karlosz — CHANGES_REQUESTED** [review]

File: src/cl-quil.lisp (hunk 1 above)
Comment (Spin1Half) [reply src/cl-quil.lisp:55]: good point!

**Review by karlosz — DISMISSED** [review]

File: src/cl-quil.lisp [hunk 3]
```diff
@@ -44,6 +47,15 @@
 ;;; generally determined by a pattern matching system, and so we have to
 ;;; consider the arity and values of parameters and arguments).
 
+(defun warn-on-ambiguous-gate-or-circuit-definition (condition)
+  "Handler which prints a warning in the presence of a CONDITION indicating an ambiguous gate or circuit declaration."
+  (check-type condition ambiguous-definition-condition)
+  (let ((instr (ambiguous-definition-instruction condition)))
+    (typecase instr
```
Comment (stylewarning) [inline src/cl-quil.lisp:54]: either make this an `etypecase` or print a diagnostic in a `t` branch

**Review by stylewarning — CHANGES_REQUESTED** [review]

**Review by stylewarning — APPROVED** [review]

**Pull request merged**